### DISCUSSION TITLE
Question on input and output from linear state space system

### DISCUSSION THREAD

**Discussion by xxyao-95** [ImperialCollegeLondon/sharpy#158] [Q&A] [answered]
Hi
From the discussion https://github.com/ImperialCollegeLondon/sharpy/discussions/137, I understand that the aeroelastic model is in non-dimentional form. Hence if I extracted the state space matrices and I run a simulation in matlab with gust velocity as input, and tip deflection as output, I need to first scale my input to make it non-dimentional and the scale my output to get the deflection in meters, is this correct?

**Comment by ngoiz** [comment, marked as answer]
That is correct, when the system is scaled it includes inputs and outputs

**Reply by xxyao-95** [reply]
Thanks!

From the Krylov ROM report, it states that for the DLTI UVLM "all displacement and velocity input need scaling in length, and the latter in time as well; the outputs scale with dynamic pressure" and "reference length is semi-chord b, and reference time is t_ref = b/u_inf"
It means that for my gust velocity input, I need to scale it with b and t_ref, the output forces from UVLM will be in non-dimentional and they will be input into the non-dimentional structure equations,
And if I want to take tip deflection as output, I need to scale it with the b again, is this correct, since deflection is only a length ?

I have run the gust response through state space in matlab and also I run the same simulation in non-linear form in SHARPy. I am comparing the results from the 2 and there is a huge difference in magnitude. I am quite confused with the scaling used in the state space system. Could you let me know if the scaling I stated above is correct?

**Reply by ACea15** [reply]
Dear xxyao,
Your definitions seem ok, have a look to the aeroelastic integration section in the following paper,
https://arc.aiaa.org/doi/abs/10.2514/1.J058153
and see if you can make sense of the theory and spot what you are missing in your MATLAB code.

**Reply by ngoiz** [reply]
Also, make sure that you are setting up your desired scaling factors through the setting in the `LinearUvlm` module (the setting [ScalingDict](https://ic-sharpy.readthedocs.io/en/master/includes/linear/assembler/linearuvlm/LinearUVLM.html)) which is shown on the [Goland Example](https://ic-sharpy.readthedocs.io/en/master/content/example_notebooks/linear_goland_flutter.html)

**Reply by xxyao-95** [reply]
Thanks for your reply! I will look into the paper.

For the ScalingDist Setting, just to double check that, I am assembleing state space at 50m/s
'ScalingDict': {'length': 0.5 * ws.c_ref,
                       'speed': u_inf,
                       'density': rho},
so  my length used is half chord, speed is 50m/s and rho is the air density at the altitude where I am simulating.
Also can I double check that the 'dt' in my 'aero_settings' should be the dt at 50m/s which is dt = chord/M/U_inf?

**Reply by ngoiz** [reply]
Yes, the `dt` in your settings is your dimensional time step. The non-dimensionalisation is performed within SHARPy. If you are having trouble getting your scalings correct it might be worth going through the code seeing how the flutter analysis of the Goland wing is performed, for example.

Starting in the `AsymptoticStability` postproc, the system is rescaled for each velocity [here](https://github.com/ImperialCollegeLondon/sharpy/blob/20f93ecdc8b67165bb7ad28f2140953ebef8cd62/sharpy/postproc/asymptoticstability.py#L255). This updates the coupled aeroelastic system at [this point](https://github.com/ImperialCollegeLondon/sharpy/blob/20f93ecdc8b67165bb7ad28f2140953ebef8cd62/sharpy/linear/assembler/linearaeroelastic.py#L258) which effectively means updating the structural equations with the new time scaling as done [here](https://github.com/ImperialCollegeLondon/sharpy/blob/20f93ecdc8b67165bb7ad28f2140953ebef8cd62/sharpy/linear/src/lingebm.py#L1250).

**Reply by xxyao-95** [reply]
Hi, I finally got the correct output from the state space system in matlab. I found out that I forgot to transpose the matrix in matlab when reading it. Here is the result.  The gust input is scaled by u_inf when it is input into the system. Thank you all for your help!
![untitled](https://user-images.githubusercontent.com/63738048/119652823-808db900-be1e-11eb-9644-ebc9d923430e.jpg)
